front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -34,14 +35,33 @@ const marginRegExValidation = /^\d+(\.\d+)?%$|^\d+(\.\d+)?min\/100km$/;
 
 const formatSuggestedViasToRowVias = (
   operationalPoints: SuggestedOP[],
-  t: TFunction<'timesStops', undefined>
-): PathWaypointColumn[] =>
-  operationalPoints?.map((op) => ({
+  pathSteps: PathStep[],
+  t: TFunction<'timesStops', undefined>,
+  startTime?: string
+): PathWaypointColumn[] => {
+  let formattedOps = [...operationalPoints];
+  const origin = pathSteps[0] as PathStep;
+  const originIndexInOps = operationalPoints.findIndex(
+    (op) =>
+      'uic' in origin &&
+      'ch' in origin &&
+      op.uic === origin.uic &&
+      op.ch === origin.ch &&
+      op.name === origin.name
+  );
+  // If the origin is in the ops and isn't the first operational point, we need to move it to the first position
+  if (originIndexInOps !== (-1 || 0)) {
+    formattedOps = formattedOps.toSpliced(originIndexInOps, 1);
+    formattedOps.unshift(operationalPoints[originIndexInOps]);
+  }
```

Comment:
Do you think it is clearer like this ?
```suggestion
  // Check if the origin is in the operational points and move it to the first position
  if (origin && 'uic' in origin && 'ch' in origin) {
    const originIndexInOps = operationalPoints.findIndex(
      (op) =>
        op.uic === origin.uic &&
        op.ch === origin.ch &&
        op.name === origin.name
    );
    if (originIndexInOps !== (-1 || 0)) {
      formattedOps = formattedOps.toSpliced(originIndexInOps, 1);
      formattedOps.unshift(operationalPoints[originIndexInOps]);
    }
  }
```

Reply:
Hmm I'm not sure because here I don't only check if origin is in the ops, I check if it's in the ops **_and_** not at the first position

Reply:
I'm ok with that, can you just move the test `'uic' in origin && 'ch' in origin` outside of the findIndex loop ? Also, I think you check if the originIndexInOps position only if the op is an uic, so you can even do it like this:
```
  if ('uic' in origin && 'ch' in origin) {
    const originIndexInOps = operationalPoints.findIndex(
      (op) => op.uic === origin.uic && op.ch === origin.ch && op.name === origin.name
    );
    // If the origin is in the ops and isn't the first operational point, we need to move it to the first position
    if (originIndexInOps !== (-1 || 0)) {
      formattedOps = formattedOps.toSpliced(originIndexInOps, 1);
      formattedOps.unshift(operationalPoints[originIndexInOps]);
    }
  }
```

Reply:
Done